### DISCUSSION TITLE
Dag is running in a infinite loop although it is deactivated

### DISCUSSION THREAD

**Discussion by kosmas9** [apache/airflow#31790] [Q&A]
Any newly created dag, even before activating it, will start running. I know this because the dag is running a very simple script that sends an email. So, even though in the GUI I see that there are no triggers and the dag itself is being deactivated, I keep receiving emails.
This happens only for newly created dags. The existing ones, when deactivated they will not run again.

Below are the logs from scheduler. Aiflow version is 2.6.1

Could you please help?

`[2023-05-22 08:59:20,191] {scheduler_job.py:182} INFO - Started process (PID=567061) to work on /opt/bitnami/airflow/dags/git-repo/dags/rebates_to_mysql/scripts/tst_script.py
[2023-05-22 08:59:20,192] {scheduler_job.py:633} INFO - Processing file /opt/bitnami/airflow/dags/git-repo/dags/rebates_to_mysql/scripts/tst_script.py for tasks to queue
[2023-05-22 08:59:20,193] {logging_mixin.py:104} INFO - [2023-05-22 08:59:20,193] {dagbag.py:487} INFO - Filling up the DagBag from /opt/bitnami/airflow/dags/git-repo/dags/rebates_to_mysql/scripts/tst_script.py
[2023-05-22 08:59:23,256] {scheduler_job.py:645} WARNING - No viable dags retrieved from /opt/bitnami/airflow/dags/git-repo/dags/rebates_to_mysql/scripts/tst_script.py
[2023-05-22 08:59:23,262] {scheduler_job.py:190} INFO - Processing /opt/bitnami/airflow/dags/git-repo/dags/rebates_to_mysql/scripts/tst_script.py took 3.074 seconds`

` 
Client:
 Version:           20.10.21
 API version:       1.41
 Go version:        go1.18.1
 Git commit:        20.10.21-0ubuntu1~18.04.3
 Built:             Thu Apr 27 05:50:21 2023
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.21
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.1
  Git commit:       20.10.21-0ubuntu1~18.04.3
  Built:            Thu Apr 27 05:36:22 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.12-0ubuntu1~18.04.1
  GitCommit:        
 runc:
  Version:          1.1.4-0ubuntu1~18.04.2
  GitCommit:        
 docker-init:
  Version:          0.19.0
  GitCommit:        `

**Comment by potiuk** [comment]
It's likely something that is run durinf parsing of the DAG not execution. You should not run anythning there. Your DAGs will be parsed regularly. Please read and digest this one  https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#top-level-python-code

**Reply by kosmas9** [reply]
Thank you for the reply. Please see my dag below. i don;t do any weird imports in the dag file. And as I mentioned there is no issue with the existing dags that have the same imports.
Could the python script be the issue? The python script exists in a different directory. But the dag does not show that it ran, so how does the python script is being executed?

`from datetime import datetime, timedelta
import pendulum
from airflow import DAG
from airflow.operators.bash_operator import BashOperator
from airflow.models import Variable


script_folder = Variable.get('DASHBOARD_REPORT_script_folder')
venv_bin_folder = Variable.get('venv_bin_folder')

default_args = {
    'owner': 'airflow',
    'depends_on_past': False,
    'start_date': datetime(2023, 5, 8, tzinfo=pendulum.timezone('Europe/Helsinki')),
    'tzinfo': pendulum.timezone('Europe/Helsinki'),
    'email_on_failure': True,
    'email_on_retry': False,
    'retries': 0,
    'retry_delay': timedelta(minutes=1)
}

dag = DAG(
    dag_id='test_2_dag',
    description='test_2_dag',
    default_args=default_args,
    schedule_interval='0 17 * * *',
    catchup=False,
    max_active_runs=1
)

main_process = BashOperator(
    task_id='test_2_dag',
    bash_command=f'{venv_bin_folder}python {script_folder}"test_2.py"',
    dag=dag,
    trigger_rule='none_skipped'
)


if __name__ == '__main__':
    dag.cli()`

**Reply by potiuk** [reply]
I have no time to look at details - but maybe somoen else can. But I can tell you that you have not read the link I sent you, because for example you are reading variables at the top level, which is clearly explained there that you should not.  Also your code is badly formatted for anyone to look at. Please look further in the direction I pointed you at.
